components/dashboard/src/feedback-form/FeedbackComponent.tsx

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2022 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { useState } from "react";
+import starryEyed from "../images/feedback/starry-eyed-emoji.svg";
+import happy from "../images/feedback/happy-emoji.svg";
+import meh from "../images/feedback/meh-emoji.svg";
+import crying from "../images/feedback/crying-emoji.svg";
+import { trackEvent } from "../Analytics";
+
+function FeedbackComponent(props: { onClose: () => void; onSubmit: () => void; isModal: boolean }) {
+    const [text, setText] = useState<string>("");
+    const [selectedEmoji, setSelectedEmoji] = useState<number | undefined>();
+
+    const emojiScore: any = {
+        starry: 4,
+        happy: 3,
+        meh: 2,
+        crying: 1,
+    };
```

Comment:
Can we avoid using `any` as the type here?

Reply:
<img src="https://media1.giphy.com/media/w89ak63KNl0nJl80ig/giphy.gif"/>

Reply:
@andrew-farries Fixed, thank you! 🤗

Reply:
One way to do this would be to use a [mapped type](https://www.typescriptlang.org/docs/handbook/2/mapped-types.html). Define:

```ts
type EmojiName = "starry" | "happy" | "meh" | "crying";
```

then

```ts
const emojiScore: { [P in EmojiName]: number } = {
        starry: 4,
        happy: 3,
        meh: 2,
        crying: 1,
    };
```

then change each call to `handleClick` to use a string literal:

```ts
onClick={(e) => handleClick("meh")}
```

Reply:
@andrew-farries Ooo, thanks! Do you prefer this way or the way I just pushed?

Reply:
You mean [3d45b5e](https://github.com/gitpod-io/gitpod/pull/10061/commits/3d45b5e68eadbf2dd656df915012282515d433ac)? 

That [fails to build](https://werft.gitpod-dev.com/job/gitpod-build-laushinka-implement-feedback-modal-9928.31) when trying to index into the `emojiScore`.

Reply:
Ahh, because of further down. Let me see.